contents/blog/feature-flags-best-practices.md

```diff
@@ -0,0 +1,198 @@
+
+None of these are laws, you can do what you want with the naming of feature flags. They are suggestions. A lot of this advice is relatively standard for naming variables in software development, feel free to take all their advice too.
+
+## 6. Minimize or group changes behind them
+
+A best practice is having a single feature flag control a single component, function, method, class, or other pieces of code.
+
+Having a flag in multiple places can be confusing. Developers expect single-use feature flags, and when they aren’t, this can cause unintended consequences. For example, a developer could remove the flag in one place without removing it in another.
+
+Multiple uses of the same feature flags increases maintenance and overhead. Every time a developer sees a feature flag, they must figure out its status and impact on code. If active, they must work around it and maintain multiple fallbacks.
+
+Having too many changes behind a feature flag makes it difficult to maintain. If you are doing experiments, it can make the relevant parts difficult to identify. It makes it more likely the parts of your feature flag break. Keeping your flags focused is best.
+
+## 7. Roll out for specific groups
+
+Feature flags are usually checked at the person level, but there are other ways. One useful and under-utilized way is rolling out based on groups.
+
+For example, a customer complains about an issue with your product. You create a fix for them but aren’t sure it will work for everyone. You can roll it out behind a feature flag and set the distribution to members of that company.
+
+![Group release](../images/blog/feature-flags-best-practices/group-release.png)
+
+Enabling flags for specific groups allows the experience to be consistent for those groups. Members of an organization aren’t seeing different UIs or getting different experiences. They also act as a “permission” to access those changes.
+
+This enables you to coordinate and communicate with those groups while rolling out features. This can be especially useful for massive features, organizations, and integrations. It allows you to beta test functionality, do [canary releases](/tutorials/canary-release), and deal with issues before a larger rollout.
+
+> **Note:** Be sure to identify users as part of that group using [group analytics](/manual/group-analytics).
+
+## 8. Remove flags at the right time
+
+Feature flags have a cost. Like any piece of code, they have maintenance associated with them. Although they are easy to create and add, figuring out when to remove them can be challenging.
+
+Leaving flags in your code for too long can confuse future developers, especially it is already rolled out and integrated. It can confuse those reading the code and the flag’s detail in PostHog.
```

Comment:
```suggestion
Leaving flags in your code for too long can confuse future developers, especially if it is already rolled out and integrated. It can confuse those reading the code and the flag’s detail in PostHog.
```